src/libostree/ostree-repo-checkout.c

```diff
@@ -582,6 +584,109 @@ checkout_file_hardlink (OstreeRepo                          *self,
   return TRUE;
 }
 
+static gboolean
+_checkout_overlayfs_whiteout_at_no_overwrite (OstreeRepoCheckoutAtOptions    *options,
+                                              int                             destination_dfd,
+                                              const char                     *destination_name,
+                                              GFileInfo                      *file_info,
+                                              GVariant                       *xattrs,
+                                              GCancellable                   *cancellable,
+                                              GError                        **error)
+{
+  guint32 file_mode = g_file_info_get_attribute_uint32 (file_info, "unix::mode");
+  if (mknodat(destination_dfd, destination_name, (file_mode & ~S_IFMT) | S_IFCHR, (dev_t)0) < 0)
```

Comment:
Don't we store the file type in the ostree mode too?  Curious why you're explicitly forcing on `S_IFCHR` here.

Reply:
In ostree it's stored as a regular file, otherwise in bare mode we would have to create character devices to get those bits preserved. Ghfn it would look more like my original approach here https://github.com/ostreedev/ostree/pull/2713 which was more complex

Reply:
or are we extracting the S_IFCHR before this call? checking,..

Reply:
ah, no, we just detect the .ostree-wh. prefix for a regular empty file, which removes the prefix and triggers this call